### ISSUE TITLE
[BUG]: King Kong Peter Jacksons - Broken Texture shuffle

### ISSUE THREAD

**Issue by lightningterror** [PCSX2/pcsx2#11870]
### Describe the Bug

King Kong Peter Jacksons experiences a broken texture shuffle effect.
Shuffle isn't being picked up from draw 127 to draw 265.

Draw 127 should be picking up RG->BA shuffle.

Extra draw info on 127:
```
m_same_group_texture_shuffle 0, tex_psm.bpp 16, memory psm bpp 16
shuffle_coords 1, rt->m_32_bits_fmt 0, src->m_32_bits_fmt 0, m_copy_16bit_to_target_shuffle 0
draw_sprite_tex 1, m_vt.m_primclass 3, TrianglesAreQuads 1
```
Both the target and source aren't picked up as 32bit, it's picking the wrong targets.
[00127_context.txt](https://github.com/user-attachments/files/17237738/00127_context.txt)
[00127_vertex.txt](https://github.com/user-attachments/files/17237739/00127_vertex.txt)

### Reproduction Steps

Run the gs dump.

### Expected Behavior

Texture shuffle effect is rendered properly.

### PCSX2 Revision

v2.1.181

### Operating System

Windows 11

### CPU

Ryzen 5 7600

### GPU

GTX 1070

### GS Settings

Default.

### Logs & Dumps

Rename .zip to .xz
[King_Kong_Peter_Jacksons_-_The_Official_Game_of_the_Movie_SLUS-21311_20230207223726.gs.zip](https://github.com/user-attachments/files/17237743/King_Kong_Peter_Jacksons_-_The_Official_Game_of_the_Movie_SLUS-21311_20230207223726.gs.zip)

**Comment by terremoth** [comment]
Yeah, I don't know if it has something to do, but last time I played I noticed some weird textures problems sometimes, if you guys need some specific testing, just let me know
